### PR TITLE
Linear Models

### PR DESCRIPTION
**Pull request by schanzer**
Take a close look at the lesson-list.txt file in pathways/algebra-2 -- note what kids WILL and WON'T have seen before this lesson!

### CODE REVIEW

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+* How did you fill in the blanks for the last question on the page?
+** (Answers will vary)
+
+@math{R^2} describes the _percentage of the variation in the y-variable that is explained by the x variable_ in our model. In other words, an @math{R^2} value of 0.20 could mean that “20% of the variation in median household income is explained by the percentage of college degrees in a state, according to our linear model”. Better models will explain a higher percentage of that variation!
+
+If the model is a perfect, the math{R^2} value will be 1.00, meaning 100% of the @math{y}-values can be explained by the @math{x}-values. Of course in the real world, no model is perfect! The @math{R^2} value for no correlation at all is *zero*. If we just drew a horizontal predictor line in the middle of the points on the scatter plot, it would mean that we expect a median income somewhere in that range but with no connection whatsoever to the percentage of people who finish college.
+
+[.lesson-instruction]
+But what if we drew a line that wasn't even in the cloud at all? Try changing your predictor to do just that.
+
```
Comment (retabak) [inline lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:198]: If you revise the worksheet as I've suggested, this question could be omitted here, I think.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@n Based on these two points, define your model in Function and Pyret Notation:
+
+[cols="^1a, ^1a", options="header"]
+|===
+| Function Notation
+| Pyret Notation
+| @math{f(x) =} @fitb{12em}{}x + @fitb{12em}{}
+| @show{(code '(define (f x) (+ (* (?ANS ______) x) (?ANS ______))))}
+|===
+
```
Comment (retabak) [inline lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc:29]: Beneath lines, add in "slope" and "y-intercept"

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+|===
+| Model (Function or Pyret Notation - whatever you prefer!)   | @math{R^2}
+|                                                             |
+|===
+
+
+@star **What does this model actually mean? Try completing the sentence below:**
+
+__This model predicts that a 1 @fitbruby{10em}{percent}{[x-axis units]} increase in @fitbruby{20em}{}{[x-axis]} is associated with a @fitbruby{20em}{@ifsoln{}}{[slope, y-units]} @fitbruby{10em}{increase}{increase / decrease} in @fitbruby{20em}{@ifsoln{}}{[y-axis]}. Based on the @math{R^2} of @fitbruby{10em}{}{@math{R^2 value}}, I would say this model fits @fitbruby{30em}{}{real well / not great / poorly / terribly}.__
+
```
Comment (retabak) [inline lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc:44]: @math is not working beneath the fill-in-the-blank line

**Review by retabak — COMMENTED** [review]
OVERALL commentary:
- During "State Demographics" and "Correlations" (the first two sections), this lesson felt like it would be more at home in a DS pathway, not an Alg2 pathway,  I think that's okay, because these sections lay necessary groundwork -- but I do think you may want to do more to highlight how they are related and relevant to "Making Predictions with Linear Models," which very much *does* feel like it belongs in an Alg2 pathway, and which seems to be the meat of the lesson.
- I recommended a structural revamp to the model-college-v-income (and a corresponding lesson restructuring) to make it really clear what all is happening - because there's a lot of stuff happening! In order to teach this, I would want/need to have the narrative that I outlined all sorted out in my head. I think we can help folks in doing that.
- Interested to learn more about your decision to offer no scaffolding / support on computing slope and y-intercept. You know more about the pathway and who we're writing it for than I do--but I suspect that many students would value some support in completing these steps.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+=== Investigate
+
+[.lesson-instruction]
+In pairs or small groups, complete @printable-exercise{pages/identifying-form.adoc}
+
+Review student answers. Some of the answers are not so clear-cut!
+
+[.lesson-instruction]
+@printable-exercise{pages/identifying-form-matching.adoc}.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:128]: Good call! Done.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+@right{@image{images/A.gif, 200}}**A relationship is strong if knowing the x-value of a data point gives us a very good idea of what its y-value will be** (knowing a student's age gives us a very good idea of what grade they're in). A strong linear relationship means that the points in the scatter plot are all clustered _tightly_ around an invisible line.
+
+@clear
+
+@right{@image{images/1a.gif, 200}}**A relationship is weak if x tells us little about y** (a student's age doesn't tell us much about their number of siblings). A weak linear relationship means that the cloud of points is scattered very _loosely_ around the line.
+
+@clear
+=== Investigate
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:120]: Done.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+- Do states with more seniors also have veterans?
+--
+
+=== Investigate
+
+[.lesson-instruction]
+- Open the @online-exercise{https://code.pyret.org/editor#share=10bWYM4rOq_ZshtX6_M-xf-bPbfz61rd1, State Demographics Starter File}
+- From the File menu, select "Save a Copy", and click "Run".
+- Working in pairs or small groups, complete @printable-exercise{exploring-states.adoc}.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:59]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+=== Common Misconceptions
+- Students often conflate strength and direction, thinking that a strong correlation _must_ be positive and a weak one _must_ be negative.
+- Students may also falsely believe that there is ALWAYS a correlation between any two variables in their dataset.
+- Students often believe that strength and sample size are interchangeable, leading to mistaken assumptions like "any correlation found in a million data points _must_ be strong!"
+
+
+=== Synthesize
+
+Have students discuss their answers to the @printable-exercise{pages/identifying-form-matching.adoc}, then turn to their Notice and Wonder.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:138]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+| Key Points For The Facilitator
+|
+* TBD
+|===
+
+== State Demographics @duration{15 minutes}
+
+=== Overview
+Students explore the @online-exercise{https://docs.google.com/spreadsheets/d/14fRmjAiCqGNI5Zr5kGSDjMavln7ILqhSX3N3zGd8CAU, State Demographics dataset}, and create scatter plots to search for relationships between columns. They share and discuss their findings with the class.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:42]: That's a good point. I'll remove the link.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+When we make a model, we want it to be the closest possible approximation of all the points. A "good fit" has most of the points very close to the line, and a "bad fit" has the points very far away.
+
+
+=== Investigate
+
+If you could draw linear model through the cloud of points in your scatter plot, where would you draw it? What would the slope and y-intercept of that model be?
+
+[.lesson-instruction]
+Complete @printable-exercise{model-college-v-income.adoc}
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:176]: LOVE this. Done.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+@n How closely does your model fit the data? Could a different line be closer to more of the points? Try changing the slope and y-intercept of your model to get a better fit, then write down the best model you came up with:
+
+@math{f(x) =} @fitbruby{15em}{}{Slope}@math{x} @hspace{2em} @math{+} @hspace{2em} @fitbruby{15em}{}{y-intercept}
+
+@n What do you think R-squared tells us? @fitb{}{}
+
+@star **What does this model actually mean? Try completing the sentence below:**
+
+__This model predicts that a 1 @fitbruby{10em}{percent}{[x-axis units]} increase in @fitbruby{20em}{college degrees}{[x-axis]} is associated with a @fitbruby{20em}{@ifsoln{$5613}}{[slope, y-units]} @fitbruby{10em}{increase}{increase / decrease} in @fitbruby{20em}{@ifsoln{median household income}}{[y-axis]}. Based on the @math{R^2} of @fitbruby{10em}{}{@math{R^2 value}}, I would say this model fits @fitbruby{30em}{}{real well / not great / poorly / terribly}.__
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc:57]: Ah, that's a limit to the racket->asciidoc->mathjax processing pipeline! I'll use an asciidoc-only workaround

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc (hunk 3 above)
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc:44]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+=== Launch
+
+--
+@right{@image{images/questions.png, 250}}
+
+- Do states with a higher percentage of college-educated people tend to be wealthier?
+- Will people commute longer for work if they make more money?
+- Do states with more seniors also have veterans?
+--
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:52]: Added some prose to address this. Resolve if this scratches the itch, feel free to modify if you know what you'd want to see, or we can talk it out tomorrow.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/exploring-states.adoc [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+@vspace{1ex}
+
+[.FillVerticalSpace, cols="^1a,^1a",options="header",stripes="none"]
+|===
+| What do you NOTICE about this dataset?
+| What do you WONDER about this dataset?
+|
+|
+|===
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/exploring-states.adoc:20]: This is a fantastic suggestion. Done.

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc (hunk 2 above)
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-your-own.adoc:29]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+**In Pyret, we can graph that function on top of the scatter plot using the `fit-model` function.** Find this function in your Contracts page!
+
+@n What is the fifth argument in the Domain? @fitb{}{}
+
+@n How do you think `fit-model` will use this argument? @fitb{}{}
+
+@fitb{}{}
+
+@n Type `fit-model(states-table, "state", "college-or-higher", "median-household-income", f)` into the Interactions Area, to see how well the model fits. Find Alaska and Alabama in the plot, then discuss the resulting display with someone else.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc:48]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+@center{@image{images/college-v-income.png}}
+
+This scatter plot appears to show a positive, linear relationship: states with higher percentages of college graduates tend to have higher median household incomes.
+
+[.lesson-instruction]
+**Suppose the United States were to add a new state**. What median household income would predict for that state, if exactly 50% of its citizens had attended college? What about 90%? 10%
+
+Let students discuss, and explain their thinking. If possible, mark off a single point for each of the hypothetical percentages, then connect those points to show a straight line.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/index.adoc:173]: Noted!

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+[cols="^1a, ^1a", options="header"]
+|===
+| Function Notation
+| Pyret Notation
+| @math{f(x) =} @fitb{10em}{@ifsoln{@math{5613.67}}}x + @fitb{10em}{@ifsoln{-83613}}
+| @show{(code '(define (f x) (+ (* (?ANS 5613.76) x) (?ANS -83613))))}
+|===
+
+@n Type it into the Definitions Area on the left, and click "Run". Make sure you fix any errors or warnings. In the Interactions Area, make sure `f(22.6)` and `f(27.5)` predict the correct median income for Alabama and Alaska!
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc:34]: Done

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+@n Try plugging in different `college-or-higher` values from _other_ states, to see how well a model based only on Alabama and Alaska fits the country.
+
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc:36]: I took this suggestion in another direction, by suggesting other states with which to _build a model_. Resolve if you're happy with this?

File: lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc [hunk 17]
```diff
@@ -0,0 +1,58 @@
+= Modeling College Degrees v. Income
+++++
```
[the remark this answers is not in the document]
Comment (schanzer) [reply lessons/Algebra2/exploring-linear-relationships/langs/en-us/pages/model-college-v-income.adoc:2]: Done

**Pull request closed without merging**